binary_search_tree/tree.py

```diff
@@ -14,42 +14,119 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def add(self, key, value = None):
```

Comment:
👍 But I'd say the space and time complexities are O(log n) if the tree is balanced and O(n) if unbalanced.